### ISSUE TITLE
Relationship to Abandoned Holon Not Handled Properly

### ISSUE THREAD

**Issue by evomimic** [evomimic/map-holons#127]
Not sure why this didn't show up in prior tests...

The `simple_abandon_staged_changes_test` case is failing on commit. See comment below for test report.

The scenario is Book holon has AUTHORED_BY relationship to Person1 and Person2.
Then Person1 is ABANDONED.
On Commit stage 1: Book and Person2 are committed, while Person 1 is (correctly) skipped.
But when committing the relationships (commit stage 2):  the Book's HolonCollection for AUTHORED_BY still contains an entry for Person1 (which is perhaps OK). But then iI think it may be trying to create a SmartLink to Person1 (which is NOT OK, because Person1 doesn't actually exist).

**Comment by evomimic** [comment]
### Test Result:

---- rstest_dance_tests::case_1_simple_abandon_staged_changes_test stdout ----
FATAL PANIC PanicInfo {
    payload: Any { .. },
    message: Some(
        DanceRequest returned 500 -- ServerError for Commit Failure 2 of 3 were successfully committed,
    ),
    location: Location {
        file: "crates/coordinator/dances/tests/shared_test/test_commit.rs",
        line: 73,
        col: 17,
    },
    can_unwind: true,
    force_no_backtrace: false,
}
thread 'rstest_dance_tests::case_1_simple_abandon_staged_changes_test' panicked at crates/coordinator/dances/tests/shared_test/test_commit.rs:73:17:
DanceRequest returned 500 -- ServerError for Commit Failure 2 of 3 were successfully committed

**Comment by dauphin3** [comment]
@evomimic I am not getting this error. pushed my flake lock. try building again.

**Comment by evomimic** [comment]
I discovered the root cause of this issue was a change to the holon's `is_accessible()` which was disallowing `read access` to holons in an `Abandoned` state. A call to get_key() from a debug statement was triggering the `is_accessble `call and the error returned was obscured because it arose at an unexpected part of the `commit()` method. I also added a warn-level message to the commit_manager's commit method so that error details are not lost if unanticipated errors are raised in the commit process.
